libs/core/algorithms/tests/unit/algorithms/all_of_tests.hpp

```diff
@@ -0,0 +1,365 @@
+//  Copyright (c) 2014-2020 Hartmut Kaiser
+//
+//  SPDX-License-Identifier: BSL-1.0
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#pragma once
+
+#include <hpx/modules/testing.hpp>
+#include <hpx/parallel/algorithms/all_any_none.hpp>
+#include <hpx/parallel/container_algorithms/all_any_none.hpp>
+#include <hpx/parallel/util/projection_identity.hpp>
+
+#include <cstddef>
+#include <iostream>
+#include <iterator>
+#include <string>
+#include <vector>
+
+#include "test_utils.hpp"
+
+///////////////////////////////////////////////////////////////////////////////
+template <typename IteratorTag>
+void test_all_of(IteratorTag)
+{
+    typedef std::vector<std::size_t>::iterator base_iterator;
```

Comment:
```suggestion
    typedef std::vector<int>::iterator base_iterator;
```

and see https://github.com/STEllAR-GROUP/hpx/pull/5639. I'm hoping disabling your CircleCI builds will get us the status checks to catch this in the future.

Reply:
Cool, thanks!, I will do the required changes and push them soon.
And I have disabled circle ci builds for my fork.